### PR TITLE
Allow compression when uploading test results

### PR DESCRIPTION
**Pull request by fische**
Adds option `test.gzip` to gzip test results before uploading them to the given URL. By default, it doesn't do it to keep things backward-compatible (even if most HTTP servers now probably support gzip compression).

### CODE REVIEW

File: src/test/xml_results.go [hunk 1]
```diff
@@ -434,9 +436,26 @@ func SerialiseResultsToXML(target *core.BuildTarget, indent bool) []byte {
...
+			return fmt.Errorf("Failed to flush gzip writer: %s", err)
+		}
+		r = &buf
+	}
+	req, err := http.NewRequest(http.MethodPost, url, r)
+	if err != nil {
+		return fmt.Errorf("Failed to create HTTP request: %s", err)
+	}
+	req.Header["Content-Type"] = []string{"application/xml"}
+	req.Header["Content-Encoding"] = []string{"gzip"}
```
Comment (goddenrich) [inline src/test/xml_results.go:457]: should this be in the if gzipped section?

File: src/core/config.go [hunk 2]
```diff
@@ -443,6 +443,7 @@ type Configuration struct {
 		Sandbox         bool         `help:"Deprecated, use sandbox.test instead."`
 		DisableCoverage []string     `help:"Disables coverage for tests that have any of these labels spcified."`
 		Upload          cli.URL      `help:"URL to upload test results to (in XML format)"`
+		Gzip            bool         `help:"True to upload the test results gzipped."`
```
Comment (goddenrich) [inline src/core/config.go:446]: Do we want to group these two as they are related?

File: src/test/xml_results.go (hunk 1 above)
Comment (fische) [reply src/test/xml_results.go:457]: 🤦 Yes it should, my bad.

File: src/core/config.go (hunk 2 above)
Comment (fische) [reply src/core/config.go:446]: If we do group them, it wont be backward compatible, so I don't think that's what we want.

Comment (peterebden) [reply src/core/config.go:446]: Might be tempted to call this `UploadGzipped` or something so it's clear that it relates to the upload (as opposed to just `test.gzip` which could mean other things)

**Review by peterebden — APPROVED** [review]
Thanks!

File: src/core/config.go (hunk 2 above)
Comment (fische) [reply src/core/config.go:446]: Ah yup good point!

**Pull request merged**